src/register/register.ts

```diff
@@ -52,6 +52,27 @@ export class Register {
       clipboard.copy(content);
     }
 
+    // Upper-case register name denotes append.
+    if (/^[A-Z]$/.test(register)) {
+     register = register.toLowerCase();
+     let existingText = Register.registers[register] && Register.registers[register].text;
+     if (existingText) {
+      if (typeof content === 'string') {
+        if (typeof existingText === 'string') {
+          content = existingText + (existingText.endsWith('\n') ? '' : '\n') + content;
+        } else {
+          content = existingText.join('\n') + '\n' + content;
```

Comment:
Yes I was unsure about this. The approach I took preserves the type of the new content, so string + string[] -> string[], string[] + string -> string.

Are you saying that if either existingText or content is string[] then the result should be string[]?